frame/balances/src/lib.rs

```diff
@@ -826,6 +831,10 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 	/// Update the account entry for `who`, given the locks.
 	fn update_locks(who: &T::AccountId, locks: &[BalanceLock<T::Balance>]) {
+		let bounded_locks = unsafe {
+			BoundedVec::<_, T::MaxLocks>::force_from(locks.to_vec(), Some("Balances Update Locks"))
+		};
+
```

Comment:
i think we shouldn't do this. what does `force_from` means here, is it "safe" to have a vec which is bigger than its bound ? If the bound is not enforced, I think we should introduce a new struct `LightBondedVec` or something like this. Which explicitly allows to go beyond the bound without unexpected behavior.

The doc of bounded vec says:
> All internal operations ensure this bound is respected.

I open an issue for this: https://github.com/paritytech/substrate/issues/8839

Reply:
@thiolliere the real solution here is to change the API of creating locks to return a result rather than always succeeding, but that is beyond the scope of this PR.

Given that we control the runtime, we actually know how many the max locks possible for a user can be, and its not attackable by a user to go past this bound unless we add pallets which would allow that.

Reply:
But yeah, i will open an issue so that this api is updated

Reply:
> Given that we control the runtime, we actually know how many the max locks possible for a user can be, and its not attackable by a user to go past this bound unless we add pallets which would allow that.

If we are sure it is correctly configured then we can do an expect with a proof here. But as long as it is not really bounded by the given configured bound, I think we should use another struct like `LightBondedVec`.